reana/cli.py

```diff
@@ -906,9 +914,15 @@ def docker_build(user, tag, component, build_arg, no_cache):  # noqa: D301
     :type no_cache: bool
     """
     components = select_components(component)
+    static_tag = tag
     for component in components:
         if is_component_dockerised(component):
             cmd = 'docker build'
+            if not static_tag:
+                tag = run_command('git describe --dirty', component,
```

Comment:
Hmm, perhaps take out the function that generates version into something like:

```
def get_current_version(srcdir, dirty=True)
```

that would behave like `get_current_branch()`?  

Could be reusable in other contexts.

(In this case you could even use `subprocess.getoutput()` so that `run_command()` would not even need the newly introduced `return_output`; though this can be handy!)